sdk/ai/azopenai/custom_client_audio.go

```diff
@@ -0,0 +1,224 @@
+		Model:          &body.Deployment,
+		Prompt:         body.Prompt,
+		ResponseFormat: body.ResponseFormat,
+		Temperature:    body.Temperature,
+	})
+
+	if err != nil {
+		return GetAudioTranslationResponse{}, err
+	}
+
+	return GetAudioTranslationResponse(resp), nil
+}
+
+func setMultipartFormData[T getAudioTranscriptionInternalOptions | getAudioTranslationInternalOptions](req *policy.Request, file []byte, options T) error {
+	body := bytes.Buffer{}
+	writer := multipart.NewWriter(&body)
+
+	writeContent := func(fieldname, filename string, src io.Reader) error {
+		fd, err := writer.CreateFormFile(fieldname, filename)
+		if err != nil {
+			return err
+		}
+		// copy the data to the form file
+		if _, err = io.Copy(fd, src); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if err := writeContent("file", "audio.mp3", bytes.NewReader(file)); err != nil {
+		return err
+	}
+
+	switch v := any(options).(type) {
+	case getAudioTranslationInternalOptions:
+		if err := writeField(writer, "model", v.Model); err != nil {
+			return err
+		}
+		if err := writeField(writer, "prompt", v.Prompt); err != nil {
+			return err
+		}
+		if err := writeField(writer, "response_format", v.ResponseFormat); err != nil {
+			return err
+		}
+		if err := writeField(writer, "temperature", v.Temperature); err != nil {
+			return err
+		}
+	case getAudioTranscriptionInternalOptions:
+		if err := writeField(writer, "language", v.Language); err != nil {
+			return err
+		}
+		if err := writeField(writer, "model", v.Model); err != nil {
+			return err
+		}
+		if err := writeField(writer, "prompt", v.Prompt); err != nil {
+			return err
+		}
+		if err := writeField(writer, "response_format", v.ResponseFormat); err != nil {
+			return err
+		}
+		if err := writeField(writer, "temperature", v.Temperature); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("failed to serialize multipart for unhandled type %T", body)
+	}
+
+	if err := writer.Close(); err != nil {
+		return err
+	}
+
+	return req.SetBody(streaming.NopCloser(bytes.NewReader(body.Bytes())), writer.FormDataContentType())
+}
+
+func getAudioTranscriptionInternalHandleResponse(resp *http.Response) (getAudioTranscriptionInternalResponse, error) {
+	at, err := deserializeAudioTranscription(resp)
+
+	if err != nil {
+		return getAudioTranscriptionInternalResponse{}, err
+	}
+
+	return getAudioTranscriptionInternalResponse{AudioTranscription: *at}, nil
+}
+
+func getAudioTranslationInternalHandleResponse(resp *http.Response) (getAudioTranslationInternalResponse, error) {
+	at, err := deserializeAudioTranscription(resp)
+
+	if err != nil {
+		return getAudioTranslationInternalResponse{}, err
+	}
+
+	return getAudioTranslationInternalResponse{AudioTranscription: *at}, nil
+}
+
+// deserializeAudioTranscription handles deserializing the content if it's text/plain
+// or a JSON object.
+func deserializeAudioTranscription(resp *http.Response) (*AudioTranscription, error) {
```

Comment:
Probably don't need to return `AudioTranscription` by pointer.

Reply:
Fixed.